lib/modules/manager/deno/schema.ts

```diff
@@ -0,0 +1,429 @@
+import validateNpmPackageName from 'validate-npm-package-name';
+import { z } from 'zod';
+import { logger } from '../../../logger';
+import {
+  getSiblingFileName,
+  localPathIsFile,
+  readLocalFile,
+} from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import {
+  Json,
+  Jsonc,
+  LooseArray,
+  LooseRecord,
+} from '../../../util/schema-utils';
+import { joinUrlParts } from '../../../util/url';
+import { extractJsrPackageName } from '../../datasource/jsr/util';
+import { id as denoVersioningId, isValid } from '../../versioning/deno';
+import type { PackageDependency, PackageFile } from '../types';
+import type { DenoManagerData } from './types';
+import { denoLandRegex, depValueRegex } from './utils';
+
+// https://github.com/denoland/deno/blob/410c66ad0d1ce8a5a3b1a2f06c932fb66f25a3c6/cli/schemas/config-file.v1.json
+export interface DenoJsonFile {
+  name?: string;
+  version?: string;
+  lock?: string | boolean | { path?: string };
+  imports?: Record<string, string>;
+  scopes?: Record<string, Record<string, string>>;
+  importMap?: string;
+  tasks?: Record<string, string | { command?: string }>;
+  compilerOptions?: {
+    types?: string[];
+    jsxImportSource?: string;
+    jsxImportSourceTypes?: string;
+  };
+  lint?: {
+    plugins?: string[];
+  };
+  workspace?: string[] | { members: string[] };
+}
```

Comment:
There is no point of using this, those fields must be parsed as schema.

Reply:
Interfaces with all-optionals are basically `any` type with nicer code completion support. While tge codebase contains a lot of this due to pre-TypeScript stage of the project, it's discouraged to introduce new types like that.